src/Connection.php

```diff
@@ -1026,10 +1026,16 @@ public function beginTransaction(): void
 
         ++$this->transactionNestingLevel;
 
-        if ($this->transactionNestingLevel === 1) {
```

Comment:
I wouldn't rearrange this code. Non-nested transactions are more common, and you just wanted to wrap the call to `$connection->beginTransaction()` in a try/catch. Let's not change anything else.

Reply:
Code structure reverted